python/ray/train/tests/test_torch_trainer.py

```diff
@@ -111,6 +129,58 @@ def test_checkpoint_freq(ray_start_4_cpus):
         trainer.fit()
 
 
+@pytest.mark.parametrize("num_gpus_per_worker", [0.5, 1, 2])
+def test_tune_torch_get_device_gpu(ray_2_node_4_gpu, num_gpus_per_worker):
+    """Tests if GPU ids are set correctly when running train concurrently in nested
+    actors (for example when used with Tune)."""
+    from ray.air.config import ScalingConfig
+    from ray.air import session
+    import time
+
+    num_samples = 2
+    num_workers = 2
```

Comment:
This should be set to `2 / num_gpus_per_worker`?

Otherwise the placement can happen anywhere on the cluster and the test can be non-deterministic?

Reply:
thanks, make sense, done